### PR TITLE
speech to text method

### PR DESCRIPTION
**Pull request by siddhant3030**
## Summary
 Google speech to text api

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [ ] Ran `mix setup` in the repository root and test.
- [ ] If you've fixed a bug or added code that is tested and has test cases. 
- [ ] In the case of adding a new API, you added a **postman** request for that.
- [ ] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

## Notes
 Please add here if any other information is required for the reviewer.

### CODE REVIEW

File: lib/glific/third_party/google_api/google_asr.ex [hunk 1]
```diff
@@ -0,0 +1,65 @@
+defmodule Glific.GoogleASR do
```
Comment (pankaj-ag) [inline lib/glific/third_party/google_api/google_asr.ex]: Let's keep the directory structure consistent 

lib/glific/third_party/ASR/google_asr.ex

File: lib/glific/third_party/google_api/google_asr.ex [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "languageCode" => "hi-IN",
+        "profanityFilter" => true
+      },
+      "audio" => %{
+        "uri" => uri
+      }
+    }
+
+    {:ok, result} = Tesla.post(new_client(org_id), url, body)
+    # IO.inspect(result)
```
Comment (pankaj-ag) [inline lib/glific/third_party/google_api/google_asr.ex]: Let's remove this comment

File: mix.exs [hunk 3]
```diff
@@ -154,7 +154,8 @@ defmodule Glific.MixProject do
       {:exvcr, "~> 0.13", only: @test_envs},
       {:dotenvy, "~> 0.1"},
       {:phoenix_swoosh, "~> 1.0"},
-      {:gen_smtp, "~> 1.1"}
+      {:gen_smtp, "~> 1.1"},
+      {:google_api_speech, "~> 0.26"}
```
Comment (pankaj-ag) [inline mix.exs]: Are we using this ?

File: lib/glific/third_party/google_api/google_asr.ex [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  @spec speech_to_text(non_neg_integer, String.t()) :: any
+  def speech_to_text(org_id, uri) do
+    url = "v1/speech:recognize"
+
+    body = %{
+      "config" => %{
+        "encoding" => "OGG_OPUS",
+        "sampleRateHertz" => 16_000,
+        "languageCode" => "hi-IN",
+        "profanityFilter" => true
```
Comment (pankaj-ag) [inline lib/glific/third_party/google_api/google_asr.ex]: can we make language configurable?

File: lib/glific/third_party/google_api/google_asr.ex [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    {:ok, result} = Tesla.post(new_client(org_id), url, body)
+    # IO.inspect(result)
+
+    case result.body["error"] do
+      nil ->
+        case result.body["results"] do
+          nil ->
+            {:error, "audio is not clear! please send it again"}
+
```
Comment (pankaj-ag) [inline lib/glific/third_party/google_api/google_asr.ex]: can you remove this deep indentation and handle by functions?

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: mix.exs (hunk 3 above)
Comment (siddhant3030) [reply mix.exs]: No. I will remove this

File: lib/glific/third_party/google_api/google_asr.ex (hunk 2 above)
Comment (siddhant3030) [reply lib/glific/third_party/google_api/google_asr.ex]: done

File: lib/glific/third_party/google_api/google_asr.ex (hunk 4 above)
Comment (siddhant3030) [reply lib/glific/third_party/google_api/google_asr.ex]: Will look into it.

File: lib/glific/third_party/asr/google_asr.ex [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        {:error, "Please check the link or Send the audio again"}
+
+      res ->
+        res |> get_in([Access.at(0), "alternatives"]) |> List.first()
+    end
+  end
+
+  @spec new_client(non_neg_integer) :: Tesla.Client.t()
+  defp new_client(org_id) do
+    token = Partners.get_goth_token(org_id, "google_cloud_storage").token
```
Comment (pankaj-ag) [inline lib/glific/third_party/asr/google_asr.ex:79]: Let's put a note here that we need to change to google_cloud_asr as a service.

**Pull request merged**